src/App/Fossa/API/BuildLink.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.API.BuildLink
+  ( getFossaBuildUrl,
+    samlUrlPath,
+  )
+where
+
+import App.Fossa.FossaAPIV1 (Organization (..), getOrganization)
+import App.Types
+import Control.Effect.Diagnostics hiding (fromMaybe)
+import Control.Effect.Lift
+import Data.Maybe (fromMaybe)
+import Data.Text (Text)
+import Data.Text.Extra (showT, underBS)
+import Fossa.API.Types (ApiOpts (..))
+import Network.HTTP.Types (urlEncode)
+import Srclib.Types (Locator (..))
+import qualified Text.URI as URI
+
+-- If SAML is not used, neither is orgID (link is created from the locator)
+defaultOrg :: Organization
+defaultOrg = Organization 0 False
+
+fossaProjectUrlPath :: Locator -> ProjectRevision -> Text
+fossaProjectUrlPath Locator {..} ProjectRevision {..} = "/projects/" <> encodedProject <> buildSelector
+  where
+    encodedProject = urlEncode' (locatorFetcher <> "+" <> locatorProject)
+    encodedRevision = urlEncode' $ fromMaybe projectRevision locatorRevision
+    -- We default to master because core does, and we need a branch to allow us to
+    -- create links directly to builds.  If this changes, then a core change should
+    -- be made allowing a link to a known revision on an unknown default branch.
+    encodedBranch = urlEncode' $ fromMaybe "master" projectBranch
+    buildSelector = "/refs/branch/" <> encodedBranch <> "/" <> encodedRevision
+
+getFossaBuildUrl :: (Has Diagnostics sig m, Has (Lift IO) sig m) => ProjectRevision -> ApiOpts -> Locator -> m Text
+getFossaBuildUrl revision apiopts locator = do
+  maybeOrg <- recover $ getOrganization apiopts
+
+  let org = fromMaybe defaultOrg maybeOrg
+      baseURI = apiOptsUri apiopts
+      relUriPath =
```

Comment:
I'd like to eliminate `defaultOrg` here -- maybe we can use a guard. something like:

```hs
relUriPath =
  case maybeOrg of
    Just org | orgUsesSaml org -> samlUrlPath org locator revision
    _ -> fossaProjectUrlPath locator revision